### PR TITLE
Improve typing in layers.points

### PR DESCRIPTION
**Pull request by dstansby**
More typing, this time low hanging fruit in `napari.layers.points`.

### CODE REVIEW

**Comment by psobolewskiPhD** [comment]
Not sure why it's just in the `pip install` test, but the error does seem relevant:
```
E   ImportError: cannot import name 'Points' from partially initialized module 'napari.layers' (most likely due to a circular import) (/opt/hostedtoolcache/Python/3.9.18/x64/lib/python3.9/site-packages/napari/layers/__init__.py)
```

File: napari/layers/points/points.py [hunk 1]
```diff
@@ -16,10 +16,12 @@
 )
 
 import numpy as np
+import numpy.typing as npt
 import pandas as pd
 from psygnal.containers import Selection
 from scipy.stats import gmean
 
+from napari.components.dims import Dims
```
Comment (psobolewskiPhD) [inline napari/layers/points/points.py]: I think this is causing the circular import error?
Should it go under `if TYPE_CHECKING:`?

**Review by psobolewskiPhD — CHANGES_REQUESTED** [review]
Other than the issue with the circular import this looks fine to me.

**Comment by psobolewskiPhD** [comment]
Pre-commit is complaining about the changes due to dropping python 3.8 support.
Could you take care of them?

To other devs: I will be mostly out the next 4 days, so once the typing is fixed, feel free to dismiss my review and proceed.
Also feel free to ping me Tuesday if nothing happens between now and then 😄

**Review by psobolewskiPhD — APPROVED** [review]

**Review by GenevieveBuckley — APPROVED** [review]
Circular import problem is resolved, everything else looks good here.
Thanks @dstansby

**Pull request merged**